openmetadata-ui/src/main/resources/ui/src/utils/CommonUtils.tsx

```diff
@@ -826,3 +827,26 @@ export const calculatePercentageFromValue = (
 ) => {
   return (value * percentageValue) / 100;
 };
+
+/**
+ * Calculates percentage from numerator and denominator with safe division
+ * @param numerator - The numerator value
+ * @param denominator - The denominator value
+ * @param precision - Number of decimal places to round to (default: 1)
+ * @returns Calculated percentage rounded to specified precision, or 0 if denominator is 0
+ * @example
+ * calculatePercentage(25, 100) // returns 25.0
+ * calculatePercentage(1, 3, 2) // returns 33.33
+ * calculatePercentage(5, 0) // returns 0 (safe division)
+ */
+export const calculatePercentage = (
+  numerator: number,
+  denominator: number,
+  precision = 1
+): number => {
+  if (denominator === 0) {
+    return 0;
+  }
+
+  return round((numerator / denominator) * 100, precision);
+};
```

Comment:
Can we move this to specific utils instead of commonUtils? It makes it difficult to reuse

Reply:
Calculating a percentage is common across applications and not related to any particular entity or feature